### PR TITLE
[TIMOB-25161] (6_2_X) iOS: Add thin, light, ultra-light & semi-bold font-sizes

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25161

### CODE REVIEW

File: iphone/Classes/WebFont.m [hunk 1]
```diff
@@ -178,16 +178,28 @@ -(UIFont *) font
                     } else {
                         font = [theFont retain];
                     }
-                } else if (self.isItalicStyle) {
-                    font = [[UIFont italicSystemFontOfSize:self.size] retain];
+                } else if ([TiUtils isIOSVersionOrGreater:@"8.2"]) {
+                    if (self.isSemiboldWeight) {
+                        font = [[UIFont systemFontOfSize:self.size weight:UIFontWeightSemibold] retain];
+                    } else if (self.isItalicStyle) {
+                        font = [[UIFont italicSystemFontOfSize:self.size] retain];
```
Comment (vijaysingh-axway) [inline iphone/Classes/WebFont.m]: Why put this api inside block of iOS 8.2 and greater ?

File: iphone/Classes/WebFont.m [hunk 2]
```diff
@@ -255,21 +267,54 @@ -(BOOL)updateWithDict:(NSDictionary *)fontDict inherits:(WebFont *)inheritedFont
...
-        } else if([fontWeightObject isEqualToString:@"bold"]){
-            didChange |= !(self.isBoldWeight)||(self.isSemiboldWeight)||(self.isNormalWeight);
-            self.isBoldWeight = YES;
-            self.isSemiboldWeight = NO;
-            self.isNormalWeight = NO;
-		} else if([fontWeightObject isEqualToString:@"normal"]){
-			didChange |= (self.isBoldWeight)||(self.isSemiboldWeight)||!(self.isNormalWeight);
+		if ([fontWeightObject isEqualToString:@"semibold"]) {
+			didChange |= !(self.isSemiboldWeight)||(self.isBoldWeight)||(self.isNormalWeight)||(self.isThinWeight)||(self.isLightWeight)||(self.isUltraLightWeight);
+    	self.isSemiboldWeight = YES;
```
Comment (vijaysingh-axway) [inline iphone/Classes/WebFont.m]: fix indentation .

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]

**Comment by hansemannn** [comment]
@vijaysingh-axway Addressed comments!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by ewieberappc — APPROVED** [review]
FR Passed.
New font weights are available and usable without issue.

**Pull request merged**